clippy_lints/src/doc/include_in_doc_without_cfg.rs

```diff
@@ -0,0 +1,35 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_opt;
+use rustc_ast::{AttrArgs, AttrArgsEq, AttrKind, AttrStyle, Attribute};
+use rustc_errors::Applicability;
+use rustc_lint::LateContext;
+use rustc_span::sym;
+
+use super::DOC_INCLUDE_WITHOUT_CFG;
+
+pub fn check(cx: &LateContext<'_>, attrs: &[Attribute]) {
+    for attr in attrs {
+        if let AttrKind::Normal(ref normal) = attr.kind
+            && normal.item.path == sym::doc
+            && let AttrArgs::Eq(_, AttrArgsEq::Hir(ref meta)) = normal.item.args
+            && !attr.span.contains(meta.span)
+            && let Some(snippet) = snippet_opt(cx, attr.span)
+            && let Some(start) = snippet.find('[')
+            && let Some(end) = snippet.rfind(']')
```

Comment:
Maybe a "we cannot remove this because a `#[doc = include]` can occupy several lines" comment would be useful here, I also checked removing these, but it doesn't seem possible =^w^=